### PR TITLE
Improve binary download page

### PR DESCRIPTION
**Pull request by mdeniz**
Refactored the view of Binaries that before was just a list of links that pointed to the details page. Now you can download the files directly or upload images to the cloud (also go to the details page as before).

new_binaries_view

BTW moved the logic on the view to the controller action.

### CODE REVIEW

**Comment by ChrisBr** [comment]
I'm missing that you remove the cloud upload from the details page, no?

**Comment by mdeniz** [comment]
> I'm missing that you remove the cloud upload from the details page, no?

Do you thing we should do it? Also then to remove the link in the title of the filename for downloading it?
The card didn't say anything about changing the details view... I can do it if we all agree. @openSUSE/open-build-service what do you think?

**Comment by ChrisBr** [comment]
> Do you thing we should do it? Also then to remove the link in the title of the filename for downloading it?
The card didn't say anything about changing the details view... I can do it if we all agree. 

The cloud upload was only temporary for the prototype so I would remove it. The download link in the header is a different thing, I would keep it for now!

**Comment by mdeniz** [comment]
Done, I removed the button from the binary view.

**Comment by mdeniz** [comment]
The hakiri warning is a false positive, because I moved the code from the controller action to another method. I don't see the problem, @ChrisBr do you mind to take a look?

**Comment by mdeniz** [comment]
WIP, needs some refactoring in the controller, please DO NOT MERGE

**Comment by hennevogel** [comment]
@mdeniz are you going to address the hakiri/codeclimate errors?

File: src/api/app/controllers/webui/package_controller.rb [hunk 1]
```diff
@@ -179,18 +175,33 @@ def binary
...
     @repository = params[:repository]
-    begin
-      @buildresult = Buildresult.find_hashed(project: @project, package: @package,
-        repository: @repository, view: ['binarylist', 'status'])
-    rescue ActiveXML::Transport::Error => e
-      flash[:error] = e.message
-      redirect_back(fallback_location: { controller: :package, action: :show, project: @project, package: @package })
+
+    results_from_backend = Buildresult.find_hashed(project: @project, package: @package, repository: @repository, view: ['binarylist', 'status'])
+    unless results_from_backend
```
Comment (hennevogel) [inline src/api/app/controllers/webui/package_controller.rb]: You will not reach this is there is an `ActiveXML::Transport::Error`...

**Comment by mdeniz** [comment]
> are you going to address the hakiri/codeclimate errors?

The hakiri is a false positive... can you mark it? Looks like I don't have permissions for it...

The code climate warnings... 2 are about number of params that we cannot change, and the other 2 are about complexity, but I thing is not that hard to understand those. What do you think?

File: src/api/app/controllers/webui/package_controller.rb (hunk 1 above)
Comment (mdeniz) [reply src/api/app/controllers/webui/package_controller.rb]: Sure, if the `Buildresult.find_hashed` call fails then you got to the rescue part on line #202. The same as before...

Comment (hennevogel) [reply src/api/app/controllers/webui/package_controller.rb]: ah true

**Pull request merged**